test/roles.spec.js

```diff
@@ -0,0 +1,377 @@
+        harvesterInstance.resource('pets', resourceSchema);
+        harvesterInstance.resource('user', resourceSchema).roles('Admin', 'SuperAdmin');
+        harvesterInstance.resource('user').getById().roles('Moderator');
+        harvesterInstance.setAuthorizationStrategy(function () {
+            return authorizationStrategy.apply(this, arguments);
+        });
+        harvesterInstance.listen(harvesterPort);
+
+        /**
+         * We need separate instance that does not have roles authorization, so that we can seed freely.
+         */
+        var seedingHarvesterInstance = harvester(config.harvester.options);
+        seedingHarvesterInstance.resource('person', resourceSchema);
+        seedingHarvesterInstance.resource('pets', resourceSchema);
+        seedingHarvesterInstance.resource('user', resourceSchema);
+        seedingHarvesterInstance.listen(harvesterPort + 1);
+        this.seedingHarvesterInstance = seedingHarvesterInstance;
+    });
+    beforeEach(function () {
+        var seederInstance = seeder(this.seedingHarvesterInstance, seedingBaseUrl);
+        return seederInstance.dropCollections('people', 'pets', 'users').then(function () {
+            return seederInstance.seedCustomFixture({people: [
+                {id: personId, name: 'Jack'}
+            ], users: [
+                {id: userId, name: 'Jill'}
+            ], pets: [
+                {id: petId, name: 'JillDog'}
+            ]});
+        });
+    });
+
+    describe('having specific config, exportRoles', function () {
+        it('should return proper roles descriptor', function () {
+            var expectedDescriptor = {
+                Admin: ['person.get', 'person.post', 'person.getById', 'person.putById', 'person.deleteById', 'person.patchById',
+                        'person.getChangeEventsStreaming', 'user.get', 'user.post', 'user.putById', 'user.deleteById', 'user.patchById',
+                        'user.getChangeEventsStreaming'],
+                SuperAdmin: ['user.get', 'user.post', 'user.putById', 'user.deleteById', 'user.patchById', 'user.getChangeEventsStreaming'],
+                Moderator: ['user.getById']
+            };
+            harvesterInstance.exportRoles().should.eql(expectedDescriptor);
+        });
+    });
+
+    describe('being authed as Admin', function () {
+        beforeEach(function () {
+            authorizationStrategy = function (request, permission, rolesAllowed) {
+                if (rolesAllowed.length === 0 || -1 < rolesAllowed.indexOf(('Admin'))) {
+                    return Promise.resolve();
+                } else {
+                    return Promise.reject();
+                }
+            }
+        });
+        it('should be allowed to get people', function (done) {
+            request(baseUrl).get('/people').expect(200).end(done);
+        });
+        it('should be allowed to post person', function (done) {
+            request(baseUrl).post('/people').send({people: [
```

Comment:
Would be really nice if you made this stuff more readable by taking advantage of newlines:
e.g:
            request(baseUrl)
               .post('/people')
               .send({people: [{name: 'Steve'}]})
               .expect(201)
               .end(done);